File: leetcode/how-many-numbers-are-smaller-than-the-current-number.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> answer;
        size_t nums_size = nums.size();
        
        for(int i : nums) {
            int min_count = 0;
            for(int j : nums){
                if(i > j)
                    ++min_count;
            }
            
            answer.push_back(min_count);
        }
        
        return answer;
    }
};

File: leetcode/how-many-numbers-are-smaller-than-the-current-number.py
from typing import List


class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        l = sorted(nums)
        d = {}

        for i, n in enumerate(l):
            if n not in d:
                d[n] = i

        res = [d[n] for n in nums]

        return res
